lib/src/model/settings/general_preferences.dart

```diff
@@ -14,6 +14,28 @@ part 'general_preferences.g.dart';
 
 const _prefKey = 'preferences.general';
 
+String? localeToJson(Locale? locale) {
+  return jsonEncode(
+    locale != null
+        ? {
+            'languageCode': locale.languageCode,
+            'countryCode': locale.countryCode,
```

Comment:
We're not using it yet, but we should take into account the `Locale.scriptCode` in the json encoder.